### ISSUE TITLE
 ManagementBackup Nemesis Error: failed to create backup target: location is not accessible

### ISSUE THREAD

**Issue by amoskong** [scylladb/scylla-cluster-tests#2188]
### Prerequisites

* [ ] Are you rebased to master ?
* [x] Is it reproducible ?
* [x] Did you perform a cursory search if this issue isn't opened ?

### Versions
* **SCT**: branch-4.1
* **scylla**: scylla-4.1.rc0-0.20200520.a1c15f06902.x86_64

### Logs
* test_id: 0e5b4b3a-e13d-48b7-b035-3d438dd48aa6
* job log: 
* job: https://jenkins.scylladb.com/view/scylla-4.1/job/scylla-4.1/job/longevity/job/longevity-2tb-4days-1Dis-2NonDis-Nemesises/2/

### Description

I touched manager backup fail in latest longevity:
- scylla-4.1.rc0-0.20200520.a1c15f06902.x86_64
- job: https://jenkins.scylladb.com/view/scylla-4.1/job/scylla-4.1/job/longevity/job/longevity-2tb-4days-1Dis-2NonDis-Nemesises/2/

ManagementBackup Nemesis fail:

```
Command: 'sudo sctool backup -c 654d1897-58ee-4cc5-b58f-ca63e4dd6ff3 --location s3:manager-backup-tests-eu-west-1 --rate-limit 30'
Exit code: 1
Stdout:
Stderr:
Error: failed to create backup target: location is not accessible
 10.0.167.169: dial tcp 10.0.167.169:10001: connect: connection refused
Trace ID: s69s_E9RQ2mCqxYURDQPWw (grep in scylla-manager logs)
```

```
$ rpm -qa |grep scylla
scylla-conf-4.1.rc0-0.20200520.a1c15f06902.x86_64
scylla-jmx-4.1.rc0-20200520.ee72ec22888.noarch
scylla-debuginfo-4.1.rc0-0.20200520.a1c15f06902.x86_64
scylla-server-4.1.rc0-0.20200520.a1c15f06902.x86_64
scylla-tools-4.1.rc0-20200520.cf56d9273ab.noarch
scylla-machine-image-4.1.rc0-20200520.a790560ae2d.noarch
scylla-python3-3.7.7-0.20200520.a1c15f06902.x86_64
scylla-tools-core-4.1.rc0-20200520.cf56d9273ab.noarch
scylla-4.1.rc0-0.20200520.a1c15f06902.x86_64
scylla-manager-agent-2.0.2-0.20200401.ab6c6b96.x86_64
scylla-kernel-conf-4.1.rc0-0.20200520.a1c15f06902.x86_64
```

### Steps to Reproduce

The problem occurred 5 times in https://jenkins.scylladb.com/view/scylla-4.1/job/scylla-4.1/job/longevity/job/longevity-2tb-4days-1Dis-2NonDis-Nemesises/2/

**Expected behavior:** [What you expected to happen]

Backup succeeded.

**Actual behavior:** [What actually happened]

Backup failed.

/CC @yarongilor @ShlomiBalalis

**Comment by fruch** [comment]
@fgelcer, that did most the work on this nemesis

**Comment by fgelcer** [comment]
@amoskong , it says that `10.0.167.169:10001` wasn't available. It could be a similar issue with the ports manager uses being taken, or either the manager-agent on `10.0.167.169` failed to start.
```
10.0.167.169: dial tcp 10.0.167.169:10001: connect: connection refused
```
The fix for the manager port issue was addressed on 2.1. On #2104 i addressed only the Prometheus port, but not the other ones, until 2.1 is released (that is planned to be soon).
If this failure happens on all executions, we could try to change the port, as we did for #2104

**Comment by fruch** [comment]
@fgelcer this doesn't seems a port issue, and it happened in all the 3 times the nemesis run.
each time to a different agent.

looking at some of the nodes log, I see there a restart of the agent each time the backup nemesis starts, could be that you are not waiting correctly for all the agent to be up before initiating the backup ?

also why you are restarting them each time, and not only once when they are installed ?

**Comment by fgelcer** [comment]
we do, because we cannot know what is the current configuration of the `scylla-manager.yaml` file when we start the test, and need to ensure the S3 location is correctly set.
in any case, there is a task to properly wait for the manager to fully init [here](https://trello.com/c/dfs94ACI/1979-add-a-waituntilmanagerup)

**Comment by ShlomiBalalis** [comment]
Solved by #2309. Closing.
